Entities/Industry/Trading/TradingPost.as

```diff
@@ -110,18 +107,41 @@ f32 onHit(CBlob@ this, Vec2f worldPoint, Vec2f velocity, f32 damage, CBlob@ hitt
 	} //no griffing
 
 	this.Damage(damage, hitterBlob);
-
+	
 	return 0.0f;
 }
 
 
 void onHealthChange(CBlob@ this, f32 oldHealth)
 {
-	CSprite @sprite = this.getSprite();
-
-	if (oldHealth > 0.0f && this.getHealth() < 0.0f)
+	// destructible in TDM and Sandbox only
+	CRules@ rules = getRules();
+	bool TDM = rules.gamemode_name == "Team Deathmatch";
+	bool SBX = rules.gamemode_name == "Sandbox";
+	
+	if (!TDM && !SBX)
+		return;
+	
+	if (oldHealth > 0.0f && this.getHealth() <= 0.0f)
 	{
-		MakeDustParticle(this.getPosition(), "Smoke.png");
-		this.getSprite().PlaySound("/BuildingExplosion");
+		// spawn trader that can be killed
+		if (this.exists("trader sex num"))
+		{
+			CBlob@ trader = server_CreateBlobNoInit("trader");
```

Comment:
You should be calling trader.Init() + doing a null check

Clients can run this code and end up hitting a null error.

![](https://fs.vamist.dev/227ece-fcXmdp9O0C8Ajhbd.png)